pkg/sidecar/param/pub_api.go

```diff
@@ -0,0 +1,238 @@
+// Copyright © 2018 One Concern
+package param
+
+import (
+	"errors"
+)
+
+type fuseParamsBundleParams struct {
+	Name         string `json:"name" yaml:"name"`
+	SrcPath      string `json:"srcPath" yaml:"srcPath"`
+	SrcRepo      string `json:"srcRepo" yaml:"srcRepo"`
+	SrcLabel     string `json:"srcLabel" yaml:"srcLabel"`
+	SrcBundle    string `json:"srcBundle" yaml:"srcBundle"`
+	DestPath     string `json:"destPath" yaml:"destPath"`
+	DestRepo     string `json:"destRepo" yaml:"destRepo"`
+	DestMessage  string `json:"destMessage" yaml:"destMessage"`
+	DestLabel    string `json:"destLabel" yaml:"destLabel"`
+	DestBundleID string `json:"destBundleID" yaml:"destBundleID"`
+	_            struct{}
+}
+
+type FUSEParams struct {
+	Globals struct {
+		SleepInsteadOfExit bool   `json:"sleepInsteadOfExit" yaml:"sleepInsteadOfExit"`
+		CoordPoint         string `json:"coordPoint" yaml:"coordPoint"`
+		Contributor        struct {
+			Name  string `json:"name" yaml:"name"`
+			Email string `json:"email" yaml:"email"`
+			_     struct{}
+		} `json:"contributor" yaml:"contributor"`
+		_ struct{}
+	} `json:"globalOpts" yaml:"globalOpts"`
+	Bundles []fuseParamsBundleParams `json:"bundles" yaml:"bundles"`
+	_       struct{}
+}
+
+type pgParamsDBParams struct {
+	Name        string `json:"name" yaml:"name"`
+	Port        int    `json:"pgPort" yaml:"pgPort"`
```

Comment:
db specs usually have a user too.

Reply:
there's a default admin user, `postgres`.

"convention over configuration", "dry", and all the usual advantages.  customizability is the marked disadvantage.

see `mock_application.sh` comments.  i could also update the readme?

i see no immediate reason to bake perms into mvp, although it is possible to wind an additional parameter through, if necessary.